src/main/java/net/snowflake/ingest/utils/IcebergDataTypeParser.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (c) 2024 Snowflake Computing Inc. All rights reserved.
+ */
+
+package net.snowflake.ingest.utils;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Lists;
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.List;
+import javax.annotation.Nonnull;
+import org.apache.iceberg.types.Type;
+import org.apache.iceberg.types.Types;
+import org.apache.iceberg.util.JsonUtil;
+
+/**
+ * This class is used to Iceberg data type (include primitive types and nested types) serialization
+ * and deserialization.
+ *
+ * <p>This code is modified from
+ * GlobalServices/modules/data-lake/datalake-api/src/main/java/com/snowflake/metadata/iceberg
+ * /IcebergDataTypeParser.java
```

Comment:
please bring over all the tests from the snowflake repo too!

Reply:
Added tests.